modules/sdk-coin-avaxp/src/lib/transaction.ts

```diff
@@ -9,14 +9,53 @@ import {
 } from '@bitgo/sdk-core';
 import { KeyPair } from './keyPair';
 import { DecodedUtxoObj, TransactionExplanation, TxData } from './iface';
-import { AddValidatorTx, AmountOutput, BaseTx, Tx, UnsignedTx } from 'avalanche/dist/apis/platformvm';
+import { AddValidatorTx, AmountOutput, BaseTx, Tx } from 'avalanche/dist/apis/platformvm';
 import { BN, Buffer as BufferAvax } from 'avalanche';
 import utils from './utils';
-import { Credential, Signature } from 'avalanche/dist/common';
+import * as createHash from 'create-hash';
+import { Credential } from 'avalanche/dist/common';
+
+// region utils to sign
+interface signatureSerialized {
+  bytes: string;
+}
+interface CheckSignature {
+  (sigature: signatureSerialized, addressHex: string): boolean;
+}
+
+function isEmptySignature(s: string): boolean {
+  return !!s && s.startsWith(''.padStart(90, '0'));
+}
+
+/**
+ * Signatures are prestore as empty buffer for hsm and address of signar for first signature.
+ * When sign is required, this method return the function that identify a signature to be replaced.
+ * @param signatures any signatures as samples to identify which signature required replace.
+ */
+function generateSelectorSignature(signatures: signatureSerialized[]): CheckSignature {
+  if (signatures.reduce((p, n) => p && isEmptySignature(n.bytes), true)) {
```

Comment:
nit, I think you are just doing this

```suggestion
  if (signatures.every((sig) => isEmptySignature(sig.bytes))) {
```

Reply:
done